crates/biome_js_analyze/tests/specs/nursery/noUselessConcat/invalid.js.snap

```diff
@@ -0,0 +1,398 @@
+---
+source: crates/biome_js_analyze/tests/spec_tests.rs
+expression: invalid.js
+---
+# Input
+```jsx
+const a = 'a' + 'b'
+const a = 'a' + 'b' + 'c'
+const a = 'a' + ('b' + 'c')
+const a = ('a' + 'b') + 'c'
+const a = foo + 'a' + 'b' + 'c'
+const a = (foo + 'a') + ('b' + 'c')
+const a = ((foo + 'a') + ('b' + 'c') + 1)
+const a = 'a' + `b`
+const a = `a` + 'b'
+const a = `a` + `b`
+const a = 'a' + 1
+const a = 1 + '1'
+const a = 1 + `1`
+const a = `1` + 1
+const a = 1 + 1 + ""
+
+```
+
+# Diagnostics
+```
+invalid.js:1:11 lint/nursery/noUselessConcat  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Useless string concatenation.
+  
+  > 1 │ const a = 'a' + 'b'
+      │           ^^^^^^^^^
+    2 │ const a = 'a' + 'b' + 'c'
+    3 │ const a = 'a' + ('b' + 'c')
+  
+  i Consider turning the expression into a single string to improve readability and runtime performance.
+  
+  i Unsafe fix: Remove the useless concatenation
```

Comment:
Why is the fix unsafe?

Reply:
While implemented this rule, I realized string concatenations can get very weird sometimes in JS, so I decided to consider the fix "unsafe" because there's a small chance that a weird concatenation pattern can result in an incorrect apply that I didn't anticipate. However, I don't know the correct approach of fix categorization in these cases, so let me know if you want me to change to safe.

Reply:
This is a safe assumption :) We could re-assess the fix category once many users have tested the rule.